extensions/mapstore/src/main/java/com/hazelcast/mapstore/GenericMapLoader.java

```diff
@@ -328,10 +344,14 @@ public Map<K, GenericRecord> loadAll(Collection<K> keys) {
 
     @Override
     public Iterable<K> loadAllKeys() {
+        // If loadAllKeys property is disabled, don't load anything
+        if (!genericMapStoreProperties.loadAllKeys) {
+            return null;
```

Comment:
Maybe  `Collections.emptyList();` to avoid NPEs?

Reply:
The calling code was handling nulls but ok, I will change it to Collections.emptyList()

Reply:
rev : 703e113888dfb36966c618c1db3d04976d47cbba